src/app/components/columns-modify-dialog.component.ts

```diff
@@ -99,12 +99,20 @@ export class ColumnsModifyDialogComponent<T extends object,O extends object> imp
 
   /**
    * Update the columns array when a checkbox is checked or unchecked
+   * checked: add the column.
+   * Unchecked: remove it
    */
   updateColumn({ checked }: MatCheckboxChange, column: ColumnKey<T, O>): void {
     if (checked) {
-      this.columns.push(column);
+      if (!this.columns.includes(column) && this.data.availableColumns.includes(column)) {
+        this.columns.push(column);
+        this.data.availableColumns = this.data.availableColumns.filter(c => c !== column);
```

Comment:
I guess it's better to give a more meaningful name to variable c in Filter method.